<?xml version="1.0"?>

<!-- Launch with command as follows: -->
<!-- roslaunch uav_id passive_flyover_experiment.launch save_id:=x -->
<!-- where x is the desired ID prefix for saved output frames -->

<launch>
  <!-- Start Gazebo with field/cow world -->
  <include file="$(find uav_id)/launch/start_world.launch">
  	<arg name="world" value="$(find uav_id)/world/test.world"/>
    <!-- Start Gazebo without visualisation/GUI (still produces simulated image topic) -->
  	<arg name="gui" value="true"/>
  </include>

  <!-- Spawn GRID-based, no-gravity camera -->
  <include file="$(find uav_id)/launch/spawn_static_camera.launch" />

  <arg name="save_id" />

  <!-- Start image processing node on image topic from UAV -->
  <node name="image_processor" pkg="uav_id" type="ImageProcessor.py" output="screen">
    <!-- Node topic remaps -->
    <remap from="cam_img" to="/downward_cam/camera/image"/>
    <!-- Node input parameters -->
  	<param name="save_input_frames" type="bool" value="false"/>
    <param name="frame_save_loc" value="$(find uav_id)/output/$(arg save_id)/"/>
    <param name="disp_raw_img_feed" type="bool" value="true"/>
  </node>

  <!-- Start the node that controls the flight of the UAV -->
  <node name="grid_camera_controller" pkg="uav_id" type="GridCameraController.py" output="screen" required="false">
    <param name="granularity" type="int" value="1"/>
  </node>

  <!-- Start rviz visualization with preset config -->
<!--   <node pkg="rviz" type="rviz" name="rviz" args="-d $(find uav_id)/rviz_cfg/will_test.rviz"/> -->
</launch>
